File: OCH_Tao/ABC/B/132.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const P = (input[1].split(" ")).map(x => parseInt(x, 10));
let cnt = 0;
for (let i = 1; i < N - 1; i++) {
  if ((P[i - 1] < P[i] && P[i] < P[i + 1]) || (P[i - 1] > P[i] && P[i] > P[i + 1])) {
    cnt++;
  }
}
console.log(cnt);

File: OCH_Tao/ABC/B/132.py
N = int(input())
P = list(map(int,input().split()))
cnt = 0
for i in range(1,N-1):
  if P[i-1]<P[i]<P[i+1] or P[i-1]>P[i]>P[i+1]:
    cnt+=1
print(cnt)